shared/core/google/storage_client.py

```diff
@@ -189,7 +194,11 @@ def download_blob_to_file(
         :param source_blob_name: The source blob name.
         :param destination_file_name: The local destination file.
         """
-        os.makedirs(os.path.dirname(destination_file_name), exist_ok=True)
+        # Handle the case where the destination_file_name is a standalone filename,
+        # such as "test.jpg", which indicates downloading to the working directory
+        # level. dir variable is then an empty string, and os.makedirs(dir) fails
+        if path_dirname_destination_file_name := os.path.dirname(destination_file_name):
```

Comment:
À choisir je pense qu'il faut mieux toujours être explicite sur le fait qu'on veut gérer ici le cas empty string, et pas None ou autre
```suggestion
        if (path_dirname_destination_file_name := os.path.dirname(destination_file_name)) != "":
```

Reply:
Perso je pense que la syntaxe `if my_string:` exprime clairement que si `my_string` est non empty (not `""`, not `None` etc) alors on veut rentrer dans la condition. Et c'est ce que je préfère dire ici, d'ailleurs si `my_string` valait potentiellement `None` dans ce contexte je voudrais aussi ne pas rentrer dans la condition. Du coup je ne vois pas de raison de préciser qu'on teste uniquement `!= ""`, d'autant que c'est un peu moins readable